### DISCUSSION TITLE
What's the correct way to access the ref used for Menu?

### DISCUSSION THREAD

**Discussion by crisshaker** [ariakit/ariakit#3976] [Questions] [answered]
How do I get access to the ref passed to Menu? I need that to implement a custom hideOnInteractiveOutside related to this: https://github.com/ariakit/ariakit/discussions/3970

<img width="421" alt="image" src="https://github.com/user-attachments/assets/c0f549e6-9d9f-45ac-866f-a3317e425c85">

**Comment by diegohaz** [comment, marked as answer]
I'm not sure I understand the question. Do you mean the element ref?

```jsx
const ref = React.useRef();

<Menu ref={ref}>
```

**Reply by crisshaker** [reply]
Ohh I was totally overthinking it then... Yeah, this works! Thanks @diegohaz

**Comment by crisshaker** [comment]
While I have your attention, do you have any thoughts on the related issue: https://github.com/ariakit/ariakit/discussions/3970
